components/brave_federated/util/synthetic_dataset.cc

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/util/synthetic_dataset.h"
+
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <random>
+
+#include "base/check.h"
+#include "base/rand_util.h"
+
+namespace brave_federated {
+
+SyntheticDataset::SyntheticDataset(std::vector<std::vector<float>> data_points)
```

Comment:
if you pass some fat container to ctor by a value (not using `const &`), them move it to a class member (not make a copy):
`data_points_(std::(data_points))`